stdlib/docs/elgamal_ecgfp5_crypto.md

```diff
@@ -0,0 +1,8 @@
+
+## std::crypto::elgamal_ecgfp5
+| Procedure | Description |
+| ----------- | ------------- |
+| encrypt_ca | Given public key, point H generated in gen_privatekey as coordinates (X,Y) on stack<br /><br />and message M, elliptic curve points (a,b) also as coordinates (X,Y) on stack<br /><br />and random scalar r on stack<br /><br />this routine computes the first elliptic curve point C_a<br /><br />C_a = r*G, G is the generator of elliptic curve<br /><br />Expected stack state<br /><br />[r0, r1, ..., r9]<br /><br />Final stack state<br /><br />[Ca_x0, ..., C_x4, Ca_y0, ..., Ca_y4, Ca_inf] |
+| encrypt_cb | Given public key, point H generated in gen_privatekey as coordinates (X,Y) on stack<br /><br />and message M, elliptic curve points (a,b) also as coordinates (X,Y) on stack<br /><br />and random scalar r on stack<br /><br />this routine computes the second elliptic curve point C_b<br /><br />C_b = M + r*H<br /><br />Expected stack state<br /><br />[H_x0, ..., H_x4, H_y0, ..., H_y4, H_inf, M_x0, ..., M_x4, M_y0, ..., M_y4, M_inf, r0, r1, ...]<br /><br />Final stack state<br /><br />[Cb_x0, ..., Cb_x4, Cb_y0, ..., Cb_y4, Cb_inf] |
+| remask_ca | Rerandomises the first half of an Elgamal ciphertext Ca<br /><br />and random scalar r to produce a rerandomised ciphertext C'a<br /><br />Expected stack state<br /><br />[Ca_x0, ..., Ca_x4, Ca_y0, ..., Ca_y4, Ca_inf, ..., r0, r1, ...]<br /><br />Final stack state<br /><br />[C'a_x0, ..., C'a_x4, C'a_y0, ..., C'a_y4, C'a_inf] |
+| remask_cb | Rerandomises the second half of an Elgamal ciphertext Cb given a public key H<br /><br />and random scalar r to produce a rerandomised ciphertext C'b<br /><br />Expected stack state<br /><br />[Cb_x0, ..., Cb_x4, Cb_y0, ..., Cb_y4, Cb_inf, H_x0, ..., H_x4, H_y0, ..., H_y4, H_inf, ..., r0, r1, ...]<br /><br />Final stack state<br /><br />[C'a_x0, ..., C'a_x4, C'a_y0, ..., C'a_y4, C'a_inf] |
```

Comment:
Looking again at the documentation I see the expected and final convention continuing to proliferate and personally it makes more sense to me. Since we already had it in that format I just left it.